Source/WebCore/css/StyleProperties.cpp

```diff
@@ -575,20 +547,79 @@ static std::optional<CSSValueID> fontStretchKeyword(double value)
 
 String StyleProperties::fontValue() const
```

Comment:
I am starting to think that maybe having a StylePropertiesSerialization (or whatnot) type to encapsulate and/or federate out the serialization logic might be a nice cleanup at some point as this file is a bit huge. (I also think we should split MutableStyleProperties and ImmutableStyleProperties into their own files).

Reply:
I agree with you. Further, there are three different things that are done with various properties, parsing, creation of computed style, and special cases for serialization of shorthands. All three have to be kept in sync for a shorthand like `font`. Better organization of this could improve things quite a bit. It never occurred to me until now, but it’s strange that for computed style we actually produce a `CSSValue` for `font`, but for specified style we build a string without ever creating a `CSSValue`.